app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+
+        $post->priority = (int)$request->get('post_priority', 0);
+        $post->word_count = (int)$request->get('word_count', 0);
+
+        $submitType = strpos($request->get('submit_type'),'下書き') !== false ? 'draft' : 'publish';
+
+        if($submitType == 'publish' && $this->currentUser->is(User::CONTRIBUTOR)){
+            return '404 Not Found';
+        }
+
+        $oldSiteId = $post->site_id;
+        $oldTargetPostId = $post->target_post_id;
+        $oldPostId = 0;
+        if($post->site_id == 0 || $oldSiteId != $request->get('site_id')) {
+            $post->site_id = $request->get('site_id');
+        }
+
+        try {
+//            $action = ($submitType == 'publish' && $post->published_status == 0 && $post->target_post_id == 0) ? 'publish' : 'edit';
+            if( ($oldSiteId != $request->get('site_id')) || $oldSiteId == 0
+                    || ($post->published_status == 0 && $post->target_post_id == 0) ) {
+                $action = 'publish';
+            } else {
+                $action = 'edit';
+            }
+
+            $post->published_status = ($submitType == 'publish' && $post->site_id > 0) ? 1 : 0;
+
+            if(!$this->currentUser->is(User::CONTRIBUTOR)
+                && $post->site_id > 0) {
+                $apiUrl = Site::find($post->site_id)->api_url;
+
+                  $returnData = (new CdsApiDriver($apiUrl, [
+                    'content' => $postContent,
+                    'title' => $postTitle,
+                    'categories' => [$post->category],
+                    'post_id' => $post->target_post_id,
+                    'post_excerpt' => $postDescription,
+                    'post_status' => $submitType,
+                    'feature_img' => $post->feature_img,
+                    'author_name' => !empty($author = User::find($post->user_id)) ? $author->username : '',
+                    'cback_url' => action('Webapp\\PostController@getEdit', $post->id),
+                    'old_post_id' => $oldPostId,
+                    'priority' => $post->priority,
+                  ]))->send($action)->getResult();
+                  if($returnData->data->post_id == 0) {
+                    throw new \Exception();
+                  }
+                  $post->content = Post::replaceIDButtonLink($returnData->data->post_id, $postContent, $oldPostId);
+                  $post->target_post_id = $returnData->data->post_id;
+                  $apiUrlOldTarget = ($oldSiteId != 0) ? Site::find($oldSiteId)->api_url : null;
+
+                  if($post->target_post_id != 0 && $oldSiteId != $post->site_id && $oldSiteId != 0 && $oldTargetPostId != 0) {
+                    $resData = (new CdsApiDriver($apiUrlOldTarget, ['post_id' => $oldTargetPostId]))->send('delete')->getResult();
+                    if($resData->data->post == 'ERROR') {
+                        throw new \Exception();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS112)